XamarinCommunityToolkit/Views/ContentButton.shared.cs

```diff
@@ -0,0 +1,173 @@
+		static void OnBorderColorPropertyChanged(BindableObject bindable, object oldValue, object newValue)
+			=> ((ContentButton)bindable).OnBorderColorPropertyChanged();
+
+		static void OnBackgroundColorPropertyChanged(BindableObject bindable, object oldValue, object newValue)
+			=> ((ContentButton)bindable).OnBackgroundColorPropertyChanged();
+
+		void OnContentTemplatePropertyChanged()
+			=> SetContent(true);
+
+		void OnContentPropertyChanged()
+			=> SetContent();
+
+		void OnCornerRadiusPropertyChanged()
+			=> Control.CornerRadius = CornerRadius;
+
+		void OnBorderColorPropertyChanged()
+			=> Control.BorderColor = BorderColor;
+
+		void OnBackgroundColorPropertyChanged()
+			=> Control.BackgroundColor = BackgroundColor;
+
+		public DataTemplate ContentTemplate
+		{
+			get => (DataTemplate)GetValue(ContentTemplateProperty);
+			set => SetValue(ContentTemplateProperty, value);
+		}
+
+		public View Content
+		{
+			get => (View)GetValue(ContentProperty);
+			set => SetValue(ContentProperty, value);
+		}
+
+		public float CornerRadius
+		{
+			get => (float)GetValue(CornerRadiusProperty);
+			set => SetValue(CornerRadiusProperty, value);
+		}
+
+		public Color BorderColor
+		{
+			get => (Color)GetValue(BorderColorProperty);
+			set => SetValue(BorderColorProperty, value);
+		}
+
+		public new Color BackgroundColor
+		{
+			get => (Color)GetValue(BackgroundColorProperty);
+			set => SetValue(BackgroundColorProperty, value);
+		}
+
+		public object CommandParameter
+		{
+			get => GetValue(CommandParameterProperty);
+			set => SetValue(CommandParameterProperty, value);
+		}
+
+		public ICommand Command
+		{
+			get => (ICommand)GetValue(CommandProperty);
+			set => SetValue(CommandProperty, value);
+		}
+
+		DataTemplate previousTemplate;
+
+		readonly object contentSetLocker = new object();
+
+		bool shouldIgnoreContentSetting;
+
+		void SetContent(bool isForceUpdate)
+		{
+			if (isForceUpdate)
+			{
+				lock (contentSetLocker)
+				{
+					shouldIgnoreContentSetting = true;
+
+					var contentFromTemplate = CreateContent();
+					if (contentFromTemplate != null)
+						Content = contentFromTemplate;
+
+					shouldIgnoreContentSetting = false;
+				}
+			}
+		}
+
+		void SetContent()
+		{
+			if (Content != null)
+				Control.Content = Content;
+				
+			if (!shouldIgnoreContentSetting)
+				SetContent(true);
+		}
+
+		View CreateContent()
+		{
+			var template = ContentTemplate;
+			while (template is DataTemplateSelector selector)
+				template = selector.SelectTemplate(BindingContext, this);
+
+			if (template == previousTemplate && Content != null)
+				return null;
+
+			previousTemplate = template;
+			return (View)template?.CreateContent();
+		}
+
+		protected override void OnControlInitialized(Frame control)
+		{
+			control.Padding = 0;
+			control.GestureRecognizers.Add(new TapGestureRecognizer
+			{
+				Command = new Command(async (parameter) =>
+				{
+					await Control.FadeTo(0.75, 100);
+					await Control.FadeTo(1, 100);
```

Comment:
should be converted to single animation with two nested ones.

Child Animation: https://www.thewissen.io/playing-with-animation-in-xamarin-forms/
(the first animation will be from 0 to .5, the second one from .5 till 1. Animation time will be 200)

Reply:
I am sure it should be converted to a single animation